File: comms/bulk-messaging/sender-pools/list-senders/list-senders.java
package com.example.usage;

import com.twilio.sdk.TwilioComms;
import com.twilio.sdk.resources.senderpools.requests.SenderPoolsListSendersRequest;
import com.twilio.sdk.types.SenderCommunicationChannel;
import com.twilio.sdk.types.SenderStatus;
import java.time.OffsetDateTime;

public class Example {
    public static void main(String[] args) {
        TwilioComms client = TwilioComms
            .builder()
            .credentials("TWILIO_ACCOUNT_SID", "TWILIO_AUTH_TOKEN")
            .build();

        client.senderPools().listSenders(
            "comms_senderpool_01h9krwprkeee8fzqspvwy6nq8",
            SenderPoolsListSendersRequest
                .builder()
                .channel(SenderCommunicationChannel.SMS)
                .status(SenderStatus.ACTIVATED)
                .startDate(OffsetDateTime.parse("2024-01-15T09:30:00Z"))
                .endDate(OffsetDateTime.parse("2024-01-15T09:30:00Z"))
                .pageToken("pageToken")
                .pageSize(50)
                .build()
        );
    }
}

File: comms/bulk-messaging/sender-pools/list-senders/list-senders.php
<?php

namespace Example;

use Twilio\Comms\TwilioComms;
use Twilio\Comms\SenderPools\Requests\SenderPoolsListSendersRequest;
use Twilio\Comms\Types\SenderCommunicationChannel;
use Twilio\Comms\Types\SenderStatus;
use DateTime;

$client = new TwilioComms(
    accountId: 'TWILIO_ACCOUNT_SID',
    authToken: 'TWILIO_AUTH_TOKEN',
);
$client->senderPools->listSenders(
    'comms_senderpool_01h9krwprkeee8fzqspvwy6nq8',
    new SenderPoolsListSendersRequest([
        'channel' => SenderCommunicationChannel::Sms->value,
        'status' => SenderStatus::Activated->value,
        'startDate' => new DateTime('2024-01-15T09:30:00Z'),
        'endDate' => new DateTime('2024-01-15T09:30:00Z'),
        'pageToken' => 'pageToken',
        'pageSize' => 50,
    ]),
);
